contrib/cloud-init/part-handler.py

```
@@ -113,15 +115,29 @@ def handle_part(data, ctype, filename, payload):
     cluster = keys.get('COMMISSAIRE_CLUSTER')
     keyfile = keys.get('ROOT_SSH_KEY_PATH', '/root/.ssh/id_rsa')
 
-    if keyfile and not os.path.isfile(keyfile):
+    if keyfile:
+        if not os.path.isfile(keyfile):
+            try:
+                subprocess.check_call(
+                    ['/usr/bin/ssh-keygen', '-q', '-N', '',
+                     '-t', 'rsa', '-f', keyfile])
+            except FileNotFoundError:
+                print('Missing /usr/bin/ssh-keygen', file=sys.stderr)
+                raise
```

I was trying to re-raise the current exception, but maybe I'm doing it wrong?

But it's really just to signal cloud-init that the script failed so it can swallow all the error details and log something generic and not very helpful.  :rage: